modules/switch/react/spec/Switch.spec.tsx

```diff
@@ -1,44 +1,123 @@
+    expect(getByRole('checkbox')).toHaveProperty('checked', true);
+  });
+
+  it('should be disabled if the disabled prop is true', () => {
+    const {getByRole} = render(<Switch disabled={true} onChange={cb} />);
+    expect(getByRole('checkbox')).toHaveProperty('disabled', true);
+  });
+
+  it('should have a reference to the input', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+
+    render(<Switch inputRef={inputRef} onChange={cb} />);
+    expect(inputRef.current!.tagName.toLowerCase()).toBe('input');
+  });
+
+  it('should have a unique id by default', () => {
+    const {getByRole} = render(<Switch disabled={true} onChange={cb} />);
+    expect(getByRole('checkbox')).toHaveProperty('disabled', true);
   });
 
-  test('Switch should spread extra props', () => {
-    const component = mount(<Switch data-propspread="test" onChange={cb} />);
-    const input = component
-      .find('input')
-      // TODO: Standardize on prop spread location (see #150)
-      .getDOMNode();
-    expect(input.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
+  it('should keep the same unique id if re-rendered', () => {
+    const {getByRole, rerender} = render(<Switch checked={false} onChange={cb} />);
+
+    const uniqueId = getByRole('checkbox').getAttribute('id');
+    expect(getByRole('checkbox')).toHaveProperty('id', uniqueId);
+
+    rerender(<Switch checked={true} onChange={cb} />);
+
+    expect(getByRole('checkbox')).toHaveProperty('checked');
+    expect(getByRole('checkbox')).toHaveProperty('id', uniqueId);
   });
 
-  test('Switch creates a unique id for each instance', async () => {
-    const fragment = mount(
-      <form>
-        <Switch checked={true} onChange={jest.fn()} disabled={false} />;
-        <Switch onChange={jest.fn()} disabled={false} />;
-      </form>
+  it('should create a unique id for each instance', () => {
+    const testIds = ['test1', 'test2'];
+    const {getByTestId} = render(
+      <>
+        <Switch data-testid={testIds[0]} onChange={cb} />
+        <Switch data-testid={testIds[1]} onChange={cb} />
+      </>
     );
 
-    const id1 = fragment
-      .find('input')
-      .at(0)
-      .getDOMNode()
-      .getAttribute('id');
+    expect(getByTestId(testIds[0]).id).not.toBe(getByTestId(testIds[1]).id);
+  });
+
+  it('should fire the onChange event handler when clicked', () => {
+    const cb = jest.fn();
+    const {getByRole} = render(<Switch checked={true} onChange={cb} />);
+
+    fireEvent.click(getByRole('checkbox'));
+    fireEvent.click(getByRole('checkbox'));
+    expect(cb).toHaveBeenCalledTimes(2);
+  });
 
-    const id2 = fragment
-      .find('input')
-      .at(1)
-      .getDOMNode()
-      .getAttribute('id');
+  it('should be a `type` of "checkbox"', () => {
+    const {getByRole} = render(<Switch onChange={cb} />);
+    expect(getByRole('checkbox')).toHaveProperty('type', 'checkbox');
+  });
+
+  it('should have a `tabIndex` of 0', () => {
```

Comment:
isn't this just testing browser functionality?

Reply:
removed, as discussed this was testing something we explicitly added to the checkbox `tabIndex=0`